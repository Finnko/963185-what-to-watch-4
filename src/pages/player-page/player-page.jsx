import React from "react";
import {getMovieById} from '../../store/reducers/movies/selectors';
import MoviePropType from '../../prop-types/movie';
import {connect} from 'react-redux';
import withVideoControls from '../../hocs/with-video-controls/with-video-controls';
import FullPagePlayer from '../../components/full-page-player/full-page-player.jsx';

const FullPagePlayerWrapped = withVideoControls(FullPagePlayer);

const PlayerPage = ({currentMovie}) => {
  const {title, videoSrc, backgroundImage} = currentMovie;

  return <FullPagePlayerWrapped
    src={videoSrc}
    title={title}
    poster={backgroundImage}
  />;
};

PlayerPage.propTypes = {
  currentMovie: MoviePropType.isRequired,
};

const mapStateToProps = (state, ownProps) => {
  const {match} = ownProps;
  const movieId = parseInt(match.params.id, 10);

  return {
    currentMovie: getMovieById(state, movieId),
  };
};

export {PlayerPage};
export default connect(mapStateToProps)(PlayerPage);
