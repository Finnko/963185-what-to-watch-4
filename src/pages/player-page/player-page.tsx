import * as React from 'react';
import {connect} from 'react-redux';
import {Redirect} from 'react-router-dom';
import {getMovieById} from '../../store/reducers/movies/selectors';
import {PathName} from '../../const';
import {Movie} from '../../interfaces';
import FullPagePlayer from '../../components/full-page-player/full-page-player';
import withVideoControls from '../../hocs/with-video-controls/with-video-controls';

const FullPagePlayerWrapped = withVideoControls(FullPagePlayer);

type Props = {
  currentMovie: Movie;
}

const PlayerPage: React.FC<Props> = (props: Props) => {
  const {currentMovie} = props;

  if (!currentMovie) {
    return <Redirect to={PathName.ROOT}/>;
  }

  const {title, videoSrc, backgroundImage} = currentMovie;

  return <FullPagePlayerWrapped
    src={videoSrc}
    title={title}
    poster={backgroundImage}
  />;
};

const mapStateToProps = (state, ownProps) => {
  const {match} = ownProps;
  const movieId = parseInt(match.params.id, 10);

  return {
    currentMovie: getMovieById(state, movieId),
  };
};

export {PlayerPage};
export default connect(mapStateToProps)(PlayerPage);
