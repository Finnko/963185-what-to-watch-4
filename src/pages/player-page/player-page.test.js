import React from 'react';
import renderer from 'react-test-renderer';
import {createMemoryHistory} from 'history';
import {Router} from 'react-router-dom';
import configureStore from 'redux-mock-store';
import NameSpace from '../../store/name-space';
import {Provider} from 'react-redux';
import {PlayerPage} from './player-page';

const mock = {
  backgroundColor: `#FDFDFC`,
  backgroundImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/Snatch.jpg`,
  description: `Unscrupulous boxing promoters, violent bookmakers, a Russian gangster, incompetent amateur robbers and supposedly Jewish jewelers fight to track down a priceless stolen diamond.`,
  director: `Guy Ritchie`,
  genre: `Comedy`,
  id: 14,
  isFavorite: true,
  title: `Snatch`,
  poster: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Snatch.jpg`,
  thumb: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/snatch.jpg`,
  preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  rating: 0.2,
  released: 2000,
  runTime: 104,
  ratingCount: 716577,
  starring: [`Jason Statham`, `Brad Pitt`, `Benicio Del Toro`],
  videoSrc: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
};


const mockStore = configureStore([]);

const store = mockStore({
  [NameSpace.MOVIES]: {
    movies: mock,
  },
});

const props = {
  currentMovie: mock,
};

describe(`PlayerPage component render correctly`, () => {
  const history = createMemoryHistory();

  it(`Should PlayerPage component render correctly`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <Router history={history}>
              <PlayerPage
                {...props}
              />
            </Router>
          </Provider>,
          {
            createNodeMock: () => {
              return {};
            }
          }
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
