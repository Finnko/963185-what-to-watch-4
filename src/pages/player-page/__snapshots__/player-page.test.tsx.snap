// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerPage component render correctly Should PlayerPage component render correctly 1`] = `
<div
  className="player"
>
  <video
    className="player__video"
    onEnded={[Function]}
    poster="img/bg-the-grand-budapest-hotel.jpg"
  />
  <div
    className="loader-wrapper"
    style={
      Object {
        "position": "fixed",
      }
    }
  >
    <div
      className="loader medium"
    />
  </div>
</div>
`;
